rootston/seat.c

```diff
@@ -220,6 +220,27 @@ static void roots_seat_init_cursor(struct roots_seat *seat) {
 	seat->cursor->request_set_cursor.notify = handle_request_set_cursor;
 }
 
+static void seat_view_destroy(struct roots_seat_view *seat_view);
+
+void roots_seat_destroy(struct roots_seat *seat) {
+	struct roots_seat_view *view, *nview;
+
+	// TODO: probably more to be freed here
+
+	wl_list_for_each_safe(view, nview, &seat->views, link) {
+		seat_view_destroy(view);
+	}
+	// Can be called from seat_destroy handler, so don't destroy seat
```

Comment:
We should destroy the seat if necessary. One way to do this is setting the seat to NULL in the destroy handler, and destroying it here only if it's non-NULL.

Reply:
This what I first tried, but doesn't work because almost everything in here depends on seat->seat being set during tear-down as well....
I think in practice this is fine, seat listens to display destroy anyway so for rootston it doesn't matter too much.

Reply:
Then make this function static, so we don't call it from outside the destroy handler.

Reply:
I can put the whole implementation in the destroy handler, and move this function back for public use that just manually calls the handler + destroys the seat?
I'm not comfortable removing a function that's been declared forever, even if it's empty and never called...